src/services/data/documents.ts

```diff
@@ -151,11 +151,13 @@ export default (
     },
 
     async unlinkGroups(schemaIdOrName, documentId, requestBody, options) {
+      const data = requestBody || {};
+
       return (
         await client.post(
           httpAuth,
           `/${schemaIdOrName}/documents/${documentId}/unlinkGroups`,
-          requestBody,
```

Comment:
I approved but I have the same hesitation as https://github.com/ExtraHorizon/javascript-sdk/pull/897#discussion_r1551541330

Reply:
I also kind of agree with Aäron it feels weirdly implicit that if an `requestBody` argument is not provided that it unlinks everything.

I do belieb that this operation should be a separate specific function and the existing behaviour for this stay the same as it would seem quite easy to make this mistake?

Reply:
Oh damn, I completely forgot we already had this discussion. I agree that there is a risk, but also interesting we're at the same discussion again. I will gladly introduce separate methods as I belief the `{}` is just bad UX / design. However, I'm currently a little stuck regarding naming this without introducing breaking changes. I'll have a think

Reply:
What about leaving the possibility like today:
```js
exh.data.documents.unlinkGroups(schemaId, documentId, {})
exh.data.documents.unlinkGroups(schemaId, documentId, { groupIds: ['58d1047f52faff0007680834'] })
```
But deprecating that ☝️
And overloading the `unlinkGroups` function with straight array usage and introducing `unlinkAllGroups`:
```js
exh.data.documents.unlinkAllGroups(schemaId, documentId)
exh.data.documents.unlinkGroups(schemaId, documentId, ['58d1047f52faff0007680834'])
```
This way it is harder to shoot yourself in the foot, but the old behavior is still possible and no weird new name needs to be found for the "specific group unlinking"-case. Traded for some extra complexity on our side dealing with the overloaded function call.